test/JsonApiDotNetCoreMongoDbExampleTests/IntegrationTests/ReadWrite/Updating/Resources/UpdateResourceTests.cs

```diff
@@ -0,0 +1,488 @@
+
+            await _testContext.RunOnDatabaseAsync(async db =>
+            {
+                var workItemInDatabase = await db.GetCollection<WorkItem>().AsQueryable()
+                    .Where(workItem => workItem.Id == existingWorkItem.Id)
+                    .FirstOrDefaultAsync();
+
+                workItemInDatabase.Description.Should().Be(newDescription);
+                workItemInDatabase.DueAt.Should().BeNull();
+                workItemInDatabase.Priority.Should().Be(existingWorkItem.Priority);
+            });
+        }
+
+        [Fact]
+        public async Task Can_update_resource_with_side_effects_with_primary_fieldset()
+        {
+            // Arrange
+            var existingWorkItem = _fakers.WorkItem.Generate();
+            var newDescription = _fakers.WorkItem.Generate().Description;
+
+            await _testContext.RunOnDatabaseAsync(async db =>
+            {
+                await db.GetCollection<WorkItem>().InsertOneAsync(existingWorkItem);
+            });
+
+            var requestBody = new
+            {
+                data = new
+                {
+                    type = "workItems",
+                    id = existingWorkItem.StringId,
+                    attributes = new
+                    {
+                        description = newDescription,
+                        dueAt = (DateTime?)null
+                    }
+                }
+            };
+
+            var route = $"/workItems/{existingWorkItem.StringId}?fields[workItems]=description,priority";
+
+            // Act
+            var (httpResponse, responseDocument) = await _testContext.ExecutePatchAsync<Document>(route, requestBody);
+
+            // Assert
+            httpResponse.Should().HaveStatusCode(HttpStatusCode.OK);
+
+            responseDocument.SingleData.Should().NotBeNull();
+            responseDocument.SingleData.Type.Should().Be("workItems");
+            responseDocument.SingleData.Id.Should().Be(existingWorkItem.StringId);
+            responseDocument.SingleData.Attributes.Should().HaveCount(2);
+            responseDocument.SingleData.Attributes["description"].Should().Be(newDescription);
+            responseDocument.SingleData.Attributes["priority"].Should().Be(existingWorkItem.Priority.ToString("G"));
+            responseDocument.SingleData.Relationships.Should().BeNull();
+
+            await _testContext.RunOnDatabaseAsync(async db =>
+            {
+                var workItemInDatabase = await db.GetCollection<WorkItem>().AsQueryable()
+                    .Where(workItem => workItem.Id == existingWorkItem.Id)
+                    .FirstOrDefaultAsync();
+
+                workItemInDatabase.Description.Should().Be(newDescription);
+                workItemInDatabase.DueAt.Should().BeNull();
+                workItemInDatabase.Priority.Should().Be(existingWorkItem.Priority);
+            });
+        }
+
+        [Fact]
+        public async Task Cannot_update_resource_for_missing_ID()
```

Comment:
This is testing logic in the serializer, so can be removed. Basically, all errors containing "Failed to deserialize request body" can be taken out.